tests/test_cu.cpp

```diff
@@ -156,7 +156,7 @@ TEST_CASE("Test zeroing cu::DeviceMemory", "[zero]") {
     stream.memcpyDtoHAsync(tgt, mem, size);
     stream.synchronize();
 
-    CHECK(data_in == data_out);
+    CHECK(static_cast<bool>(!memcmp(data_in.data(), data_out.data(), size)));
```

Comment:
This was actually correct, see [here](https://en.cppreference.com/w/cpp/container/vector/operator_cmp). Therefore:
```suggestion
    CHECK(data_in == data_out);
```